packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx

```diff
@@ -239,14 +260,24 @@ function ChartsXAxis(inProps: ChartsXAxisProps) {
               />
             )}
 
-            {formattedValue !== undefined && !skipLabel && showTickLabel && (
+            {needsMeasuring || (formattedValue !== undefined && isLabelVisible && showTickLabel) ? (
               <TickLabel
                 x={xTickLabel}
                 y={yTickLabel}
                 {...axisTickLabelProps}
-                text={formattedValue.toString()}
+                measuring={needsMeasuring}
```

Comment:
Another option would be to merge the styles and pass `visibility: 'hidden'`. Not sure what would be better 🤔